addons/account/account.py

```diff
@@ -278,6 +286,8 @@ def write(self, vals):
             if ('company_id' in vals and journal.company_id.id != vals['company_id']):
                 if self.env['account.move'].search([('journal_id', 'in', self.ids)], limit=1):
                     raise UserError(_('This journal already contains items, therefore you cannot modify its company.'))
```

Comment:
Did you target the right line ? I don't see the link between required bank account for bank journals and not allowing to change the company of a journal that has been used